entity.go

```diff
@@ -101,15 +101,29 @@ func (lr *limitedReader) Read(p []byte) (int, error) {
 	return n, err
 }
 
-// Read reads a message from r. The message's encoding and charset are
-// automatically decoded to raw UTF-8. Note that this function only reads the
-// message header.
-//
-// If the message uses an unknown transfer encoding or charset, Read returns an
-// error that verifies IsUnknownCharset or IsUnknownEncoding, but also returns
-// an Entity that can be read.
-func Read(r io.Reader) (*Entity, error) {
-	lr := &limitedReader{R: r, N: maxHeaderBytes}
+type ReadOptions struct {
+	// MaxHeaderBytes limits the maximum permissible size of a message header
+	// block. If exceeded, an error will be returned.
+	// -1 == no limit, 0 == default value (1MB)
+	MaxHeaderBytes int64
+}
+
+// checkValues will modify the ReadOptions struct setting defaults or special
+// values. Must be called before the ReadOptions struct is used.
+func (o *ReadOptions) checkValues() {
```

Comment:
checkValues seems like an awkward name, considering that you aren't checking values (meaning returning a bool or error or some such) but modify them.

Reply:
I agree, it wasn't easy to name. I initially had `setDefaults()`, but thought that might be confused with a constructor type func. Also considered `validate()` but that has the same issues as `checkValues()`. `setDefaults()` is probably the best fit, but I'm open to other names or ways to approach this.

Reply:
Perhaps another approach would be to take a pointer receiver like you suggested, but have a func `readerOptionsFromUserOpts(userOpts *ReaderOptions) (*ReaderOptions)` (or something else, I'm terrible at short names), which would just make a new instance that is guaranteed valid?

Reply:
Yeah, sounds good. `ReaderOptions.withDefaults` may be a reasonable name?